packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -1551,11 +1551,28 @@ def initial_call(self, builder: LROOperationType) -> List[str]:
 
     def return_lro_poller(self, builder: LROOperationType) -> List[str]:
         retval = []
+        lro_options = (
+            {
+                "final-state-via": builder.lro_options["final-state-via"],
+            }
+            if builder.lro_options and "final-state-via" in builder.lro_options
+            else {}
+        )
+        if self.code_model.options["version_tolerant"]:
+            try:
+                api_version = next(
+                    p
+                    for p in builder.parameters.parameters
+                    if p.client_name == "api_version" and p.client_default_value
```

Comment:
is this the same logic we use for checking if a client api version param exists in paging?